lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -272,6 +273,11 @@ def _set_scale(self):
         if self._orientation == 'y':
             pscale = self._parent.yaxis.get_scale()
             set_scale = self.set_yscale
+        if pscale == self._parentscale:
```

Comment:
do we want `==` or `is` here?

Reply:
It's ==, get_scale returns a string.  Well really we should retrieve the scale object rather than a scale string, but that's a separate issue...